core/services/ocr2/plugins/median/services.go

```diff
@@ -113,18 +113,30 @@ func NewMedianServices(ctx context.Context,
 		}
 	}
 
-	dataSource, juelsPerFeeCoinSource := ocrcommon.NewDataSourceV2(pipelineRunner,
+	dataSource := ocrcommon.NewDataSourceV2(pipelineRunner,
 		jb,
 		*jb.PipelineSpec,
 		lggr,
 		runSaver,
-		chEnhancedTelem,
-	), ocrcommon.NewInMemoryDataSource(pipelineRunner, jb, pipeline.Spec{
+		chEnhancedTelem)
+
+	juelsPerFeeCoinSource := ocrcommon.NewInMemoryDataSource(pipelineRunner, jb, pipeline.Spec{
 		ID:           jb.ID,
 		DotDagSource: pluginConfig.JuelsPerFeeCoinPipeline,
 		CreatedAt:    time.Now(),
 	}, lggr)
 
+	if pluginConfig.JuelsPerFeeCoinCaching {
+		lggr.Infof("juelsPerFeeCoinSource caching is enabled for jb %s, cache expiration is set to %s", jb.Name.ValueOrZero(), pluginConfig.JuelsPerFeeCoinCacheDuration.String())
+		juelsPerFeeCoinSource = ocrcommon.NewInMemoryDataSourceCache(pipelineRunner, jb, pipeline.Spec{
```

Comment:
why does the cache need such a complex signature? instead of wrapping the pipeline i think would either change the datasource to take an optional caching configuration or i would have wrapped the datasource directly ~

juelsPer... = NewCachableDS(juelPer...)
or
juelsPer.. = NewInMemoryDataSource(a, b, cfg)

cfg = struct {
x
y
cacheOpts
}